### PR TITLE
prov/rxm: Don't call fi_cancel on closed msg endpoint.

### PR DESCRIPTION
**Pull request by a-ilango**
- Also report correct completion flags.
- Add a man page note about an issue with fi_cq_sread / wait objects.

### CODE REVIEW

File: man/fi_rxm.7.md [hunk 1]
```diff
@@ -88,6 +88,11 @@ RxM provider does not support the following features:
 RxM provider should work fine for client - server programs like fabtests. Support for MPI, SHMEM
 and other applications is work in progress.
 
+## Known issues
+
+Applications that use fi_cq_sread or wait on CQ fd may not function correctly. This would be fixed
+in the next release.
```
Comment (shefty) [inline man/fi_rxm.7.md]: Can you add an assert or warning print statement to sread to help notify a developer of the problem?

File: prov/rxm/src/rxm_cq.c [hunk 2]
```diff
@@ -130,6 +156,9 @@ static int rxm_finish_send_nobuf(struct rxm_tx_entry *tx_entry)
 					"Unable to report completion\n");
 			return ret;
 		}
+#if ENABLE_DEBUG
+		rxm_cq_log_comp(tx_entry->comp_flags);
+#endif
```
Comment (shefty) [inline prov/rxm/src/rxm_cq.c]: nit: a cleaner way of doing this is to move the ENABLE_DEBUG around the call definition, and provide an empty macro in the else case.

File: man/fi_rxm.7.md (hunk 1 above)
Comment (a-ilango) [inline man/fi_rxm.7.md]: I could return -FI_ENOSYS for the cq_sread/sreadfrom calls until they are fixed.

**Comment by a-ilango** [comment]
Thanks. I've updated the PR with suggested changes.

**Pull request merged**